datalad/nonasyncrunner.py

```diff
@@ -0,0 +1,162 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Wrapper for command and function calls, allowing for dry runs and output handling
+
+"""
+
+import logging
+import os
+import queue
+import subprocess
+import threading
+import time
+from typing import Any
```

Comment:
xoxoxo -- I thought to complain with "as much as I would have liked to add typing, let's postpone for this PR: it would impact CLI startup time notably" but apparently we were impacted already for a while due to `asyncio` itself (at least with python3 3.9.1):

```
  File "/home/yoh/proj/datalad/datalad-master/datalad/__init__.py", line 47, in <module>
    from .config import ConfigManager
  File "/home/yoh/proj/datalad/datalad-master/datalad/config.py", line 22, in <module>
    from datalad.cmd import (
  File "/home/yoh/proj/datalad/datalad-master/datalad/cmd.py", line 21, in <module>
    import asyncio
  File "/usr/lib/python3.9/asyncio/__init__.py", line 8, in <module>
    from .base_events import *
  File "/usr/lib/python3.9/asyncio/base_events.py", line 45, in <module>
    from . import staggered
  File "/usr/lib/python3.9/asyncio/staggered.py", line 6, in <module>
    import typing
  File "/tmp/badimport/typing.py", line 1, in <module>
    raise ImportError
ImportError
```

So finally we can resurrect https://github.com/datalad/datalad/pull/4513 and see other issue referenced therein, and introduce type hints through out the code!!! 

FWIW -- I verified that `import datalad` and `datalad --help-np` aren't notably impacted by this PR for me since we already tolerated import of typing via asyncio , so I would just welcome its addition here and elsewhere ;-)